service/src/main/java/org/cbioportal/genome_nexus/service/internal/EnsemblServiceImpl.java

```diff
@@ -19,23 +20,55 @@
     }
 
     @Override
-    public List<EnsemblTranscript> getEnsemblTranscriptsByTranscriptId(String transcriptId) {
-        return this.ensemblRepository.findByTranscriptId(transcriptId);
+    public List<EnsemblTranscript> getEnsemblTranscriptsByTranscriptId(String transcriptId)
+        throws EnsemblTranscriptNotFoundException
+    {
+        List<EnsemblTranscript> transcripts = this.ensemblRepository.findByTranscriptId(transcriptId);
+
+        if (transcripts.size() == 0) {
```

Comment:
You are probably using a path variable where you should be using a query variable, otherwise you wouldn't be returning 404 from a list-returning endpoint.

Reply:
@ersinciftci Thanks a ton for checking! This is a very good point and made clear that we actually had to change the endpoints quite drastically. Much appreciated!

Reply:
@inodb No problem! Btw I realized I worded the comment poorly, list-returning endpoints can return 404 in some cases, like `/studies/{studyId}/samples` would return 404 if studyId is invalid, but it wouldn't have a `samples.size() == 0` check for it, it would have `study == null`, that's what I meant. @onursumer